src/UKHO.ERPFacade.Common/Services/ERPFacadeService.cs

```diff
@@ -58,20 +58,21 @@ public List<UnitsOfSalePrices> MapAndBuildUnitsOfSalePrices(List<PriceInformatio
                             {
                                 var existingUnitOfSalePrice = unitsOfSalePriceList.Where(x => x.UnitName.Contains(priceInformation.ProductName)).FirstOrDefault();
 
-                                var effectiveUnitOfSalePriceDurations = existingUnitOfSalePrice.Price.Where(x => x.EffectiveDate.ToString("yyyyMMdd") == priceInformation.EffectiveDate).ToList();
+                                var effectiveUnitOfSalePriceDurations = existingUnitOfSalePrice!.Price.Where(x => x.EffectiveDate.ToString("yyyyMMdd") == priceInformation.EffectiveDate).ToList();
```

Comment:
can we define the date format as below and use it wherever required ? 
private const string EffectiveDateFormat = "yyyyMMdd";